rhizome/host/lib/vm_setup.rb

```diff
@@ -444,9 +446,13 @@ def download_boot_image(boot_image, custom_url: nil)
         end
       end
 
-      # Images are presumed to be atomically renamed into the path,
-      # i.e. no partial images will be passed to qemu-image.
-      r "qemu-img convert -p -f #{initial_format.shellescape} -O raw #{temp_path.shellescape} #{image_path.shellescape}"
+      if initial_format == "raw"
+        r "mv #{temp_path.shellescape} #{image_path.shellescape}"
```

Comment:
Consider `File.rename`. Has a lot less going on, more or less hits the syscall only rather than forking through the shell and into `mv`, which in turn will wind its way around to calling the rename syscall. The main thing I can think of `mv` gives you is a file copy if the source and target are on two file systems (yeah, the manual page is a cornucopia of features, but this is the one most likely to surprise someone)

Reply:
Good insights on `File.rename` and `mv`. I changed it.

Reply:
It looks like we can't use `File.rename`, because we are moving between different disk partitions (btrfs ➔ ext4) @fdr 

> /home/rhizome/host/lib/vm_setup.rb:451:in `rename': Invalid cross-device link @ rb_file_s_rename - (/tmp/github-ubuntu-2204.raw.tmp, /var/storage/images/github-ubuntu-2204.raw) (Errno::EXDEV)

Reply:
@fdr I reverted back it to `mv`

Reply:
question: is this bad? because...mv that is actually copy-and-delete is, well, slow.  and images are big. Are we staging images in the wrong place altogether, and should use File.rename as well?

Reply:
We may consider downloading the theme image to `/var/storage/images` instead of `/tmp`. This would place them in the same partition. The only advantage I can envision is that if the download fails for any reason, `/tmp` automatically cleaned after a certain period. However, we would need to manage the cleanup process ourselves. 

There's already an existing issue related to this: https://github.com/ubicloud/ubicloud/issues/911. Our first step might be to implement a cleanup logic.

Reply:
I think we should definitely avoid copy+delete of 86GB between file systems and stick with the rename system call (`File.rename`). 

I think we can write not-quite-/tmp cleaning routines that work well...really, mostly better. The main thing to watch out for is spinning and producing a lot of partially complete files. It is this reason I don't use mktmp but instead compute a static name somehow.

I think with a bit of careful thought, we can avoid issues like two processes writing stuff to a single file, even in the presence of program bugs.

Reply:
What are your thoughts on these two PRs?
https://github.com/ubicloud/ubicloud/pull/1140
https://github.com/ubicloud/ubicloud/pull/1143